cmd/agent/subcommands/dogstatsdcapture/command.go

```diff
@@ -27,50 +32,52 @@ import (
 	"google.golang.org/grpc/metadata"
 )
 
-var (
-	dsdCaptureDuration   time.Duration
-	dsdCaptureFilePath   string
-	dsdCaptureCompressed bool
-)
-
 const (
 	defaultCaptureDuration = time.Duration(1) * time.Minute
 )
 
+// cliParams are the command-line arguments for this subcommand
+type cliParams struct {
+	*command.GlobalParams
+
+	dsdCaptureDuration   time.Duration
+	dsdCaptureFilePath   string
+	dsdCaptureCompressed bool
+}
+
 // Commands returns a slice of subcommands for the 'agent' command.
 func Commands(globalParams *command.GlobalParams) []*cobra.Command {
+	cliParams := &cliParams{
+		GlobalParams: globalParams,
+	}
+
 	dogstatsdCaptureCmd := &cobra.Command{
 		Use:   "dogstatsd-capture",
 		Short: "Start a dogstatsd UDS traffic capture",
 		Long:  ``,
 		RunE: func(cmd *cobra.Command, args []string) error {
-
-			err := common.SetupConfigWithoutSecrets(globalParams.ConfFilePath, "")
-			if err != nil {
-				return fmt.Errorf("unable to set up global agent configuration: %v", err)
-			}
-
-			err = config.SetupLogger(config.CoreLoggerName, config.GetEnvDefault("DD_LOG_LEVEL", "off"), "", "", false, true, false)
-			if err != nil {
-				fmt.Printf("Cannot setup logger, exiting: %v\n", err)
-				return err
-			}
-
-			return dogstatsdCapture()
+			return fxutil.OneShot(dogstatsdCapture,
+				fx.Supply(cliParams),
+				fx.Supply(core.BundleParams{
+					ConfFilePath:      globalParams.ConfFilePath,
+					ConfigLoadSecrets: false,
+				}.LogForOneShot("CORE", "off", true)),
+				core.Bundle,
+			)
 		},
 	}
 
-	dogstatsdCaptureCmd.Flags().DurationVarP(&dsdCaptureDuration, "duration", "d", defaultCaptureDuration, "Duration traffic capture should span.")
-	dogstatsdCaptureCmd.Flags().StringVarP(&dsdCaptureFilePath, "path", "p", "", "Directory path to write the capture to.")
-	dogstatsdCaptureCmd.Flags().BoolVarP(&dsdCaptureCompressed, "compressed", "z", true, "Should capture be zstd compressed.")
+	dogstatsdCaptureCmd.Flags().DurationVarP(&cliParams.dsdCaptureDuration, "duration", "d", defaultCaptureDuration, "Duration traffic capture should span.")
+	dogstatsdCaptureCmd.Flags().StringVarP(&cliParams.dsdCaptureFilePath, "path", "p", "", "Directory path to write the capture to.")
+	dogstatsdCaptureCmd.Flags().BoolVarP(&cliParams.dsdCaptureCompressed, "compressed", "z", true, "Should capture be zstd compressed.")
 
 	// shut up grpc client!
 	grpclog.SetLoggerV2(grpclog.NewLoggerV2(ioutil.Discard, ioutil.Discard, ioutil.Discard))
 
 	return []*cobra.Command{dogstatsdCaptureCmd}
 }
 
-func dogstatsdCapture() error {
+func dogstatsdCapture(log log.Component, config config.Component, cliParams *cliParams) error {
```

Comment:
This function is requiring components, and we require those two components to be started.  This is presently a little awkward, because this subcommand _only_ relies on these components to initialize logging and config, and not to actually log messages or fetch config parameters.  In the next phases of  componentization, the values will be used.